src/main/java/org/jabref/preferences/JabRefPreferences.java

```diff
@@ -1392,6 +1399,33 @@ public GroupsPreferences getGroupsPreferences() {
         return groupsPreferences;
     }
 
+    private void storeStyle(OOStyle style) { //where to call?
+        if (style instanceof CSLStyle cslStyle) { // why use iscitationstylefile?
+            OO_CURRENT_STYLE = cslStyle.getPath();
+        } else if (style instanceof JStyle jStyle) {
+            OO_CURRENT_STYLE = jStyle.getName();
+        }
+    }
+
+    public OOStyle getStyle() {
+        if (style != null) {
+            return style;
+        }
+
+        String currentStyle = get(OO_CURRENT_STYLE);
```

Comment:
Here, you now need to do a check based on the value in the preferences:
If it ends with .csl  then you have a CSL style 
  if (CitationStyle.isCitationStyleFile(currentStyle)) {   
CitationStyle.createCitationStyleFromFile(currentStyle)

if( isJStyle(currentStyle) )
ooBibStyleFromjstyle.

Reply:
Okay, makes sense now. I tried to store the styletype object property, whereas I had to use a string and deduce the type from that.